heron/api/src/java/com/twitter/heron/streamlet/impl/StreamletImpl.java

```diff
@@ -125,6 +125,16 @@ public String getName() {
     return name;
   }
 
+  protected void setDefaultNameIfNone(String prefix, Set<String> stageNames) {
```

Comment:
I think definition of Streamlet/StagePrefixNameType as enum can be useful by supporting type-safety and all types can sit in single place.